lib/schema.js

```diff
@@ -81,14 +81,15 @@ Schema.prototype.create = function(obj, schema){
 			}
 
 			if (propertySchema.type === 'Array') {
-				return Promise.each(value, function(v, i) {
-					this.i = i;
-					return propertySchema.of.create({subSchema: v});
+				return Promise.map(value, function(v, i) {
```

Comment:
The **(v,i)** arguments seem a little cryptic to me. Could you explain what they stand for?